components/service/telemetry/src/main/java/org/mozilla/telemetry/ping/TelemetryPingBuilder.java

```diff
@@ -30,9 +30,13 @@ public TelemetryPingBuilder(@NonNull TelemetryConfiguration configuration, @NonN
         this.type = type;
         this.measurements = new LinkedList<>();
 
-        // All pings contain a version and a client id
+        // All pings contain a version and a client id (with exception below)
         addMeasurement(new VersionMeasurement(version));
-        addMeasurement(new ClientIdMeasurement(configuration));
+
+        // Fire-tv pocket telemetry ping should not include client-id (see #1606)
+        if (!type.equals(TelemetryPocketEventPingBuilder.TYPE)) {
+            addMeasurement(new ClientIdMeasurement(configuration));
+        }
```

Comment:
Instead of requiring `TelemetryPingBuilder` to know about ping types that extend it.. how about introducing something like `shouldIncludeClientId()` which returns `true` by default and is overriden by `TelemetryPocketEventPingBuilder` to return `false`?